qmetaobject/src/lib.rs

```diff
@@ -881,6 +881,87 @@ pub fn install_message_handler(logger: extern "C" fn(QtMsgType, &QMessageLogCont
     cpp!(unsafe [logger as "QtMessageHandler"] { qInstallMessageHandler(logger); })
 }
 
+/// Embed files and made them available to the Qt resource system.
+///
+/// The macro accepts an identifier with optional preceding visibility modifier,
+/// and a comma-separated list of resources. Then macro generates a function
+/// with given name and visibility, which can be used to register all the
+/// resources.
+///
+/// # Input
+///
+/// The macro accepts:
+///  - an identifier with optional preceding visibility modifier
+///  - and a comma-separated list of _resources_.
+///
+/// Each _resource_ consists of a
+///  - prefix path (corresponds to qrc format)
+///  - optional base directory path after an `as` keyword
+///    (custom extention which does not interfere with qrc format,
+///    but merely act as if `cd path/to/dir` was invoked before processing
+///    the resource)
+///  - and a braced list of comma-separated _files_.
+///
+/// Each _file_ is specified as
+///  - a path to the file (corresponds to qrc format)
+///  - with an optional alias path after an `as` keyword (corresponds to qrc format).
+///
+/// The file paths are relative to the directory in which cargo runs plus the
+/// optional base directory path of a particular resource.
+///
+/// It does not matter if the prefix has leading '/' or not.
+///
+/// # Output
+///
+/// The macro creates a function with given name and visibility modifier,
+/// that needs to be run in order to register the resource. Calling the
+/// function more than once has no effect.
+///
+/// # Example
+///
+/// Consider this project files structure:
+/// ```text
+/// .
+/// ├── Cargo.toml
+/// └── tests/qml
+/// │   ├── qml.qrc
+/// │   ├── main.qml
+/// │   └── Bar.qml
+/// └── src
+///     └── main.rs
+/// ```
+/// then the following Rust code:
+/// ```
+/// # extern crate qmetaobject;
+/// # use qmetaobject::qrc;
+/// qrc!(my_resource,
+/// # // For maintainers: this is actually tested against read files.
+///     "foo" as "tests/qml" {
```

Comment:
should it not be the opposite order: `"test/qml" as "foo"`  ?
Makes more sense to have what's after the "as" to be the virtual thing in the ressource.

Reply:
Makes sense. My bad. Was sleepy.

So, the grammar should be like this:

Resource: `[ <physical> as ] <virtual> { File* }`
File: `<physical> [ as <virtual> ]`

The opposite optional parts.